wordpress-mobile/WordPress-iOS
# Media: Add support for remaining asset types

Part of https://github.com/wordpress-mobile/WordPress-iOS/issues/21457

There are multiple changes in this PR, so please bear with me. Once this is merged, the new screens will cover 100% of the original MediaLibraryViewController functionally (and a little bit more). I've been commenting out the re-implement lines as I was working on this project, so this is how I know.

Changes:

- Add support for remaining document types: video, document, audio
- Add GIF rendering and increase the scale of requested GIFs to the max of 2x. Downscaling too much resulted in artifacts during playback.
- Add accessibility support
- Pop details view controller if media is deleted

I have a couple of more performance improvements in mind, but these will be done later.

<img width="320" alt="Screenshot 2023-09-29 at 5 30 49 PM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/2b2be5b1-597a-4994-afd5-24e67950d831">   <img width="320" alt="Screenshot 2023-09-29 at 5 30 51 PM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/5716529a-95b3-4d34-9c4d-98aca6b44440">

## To test:

- Verify that the existing video assets display the duration
- Verify that when you upload a new video, it displays the duration
- Verify that documents and audio have a respective icon and their filenames displayed
- Verify that cells are accessibility elements (copied from the previous version)
- Verify that after going to the details page and deleting the asset on it, the details page gets popped from the stack (see [recording](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/5c88d4ba-4ced-4cb0-b5fb-0a2dfac06cbb))
- Verify that GIFs are now animated (we already had the files, so I thought, why not just animate them; it wasn't supported before)

## Regression Notes
1. Potential unintended areas of impact: n/a (no production code changed)
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:


UI Changes testing checklist:
